src/custom/utils/operator/index.ts

```diff
@@ -191,31 +188,54 @@ export async function getPriceQuote(params: PriceQuoteParams): Promise<PriceInfo
   const [checkedBaseToken, checkedQuoteToken] = [checkIfEther(baseToken, chainId), checkIfEther(quoteToken, chainId)]
   console.log('[util:operator] Get Price from API', params)
 
-  return _getJson(
-    chainId,
-    `/markets/${toApiAddress(checkedBaseToken, chainId)}-${toApiAddress(checkedQuoteToken, chainId)}/${kind}/${amount}`,
-    'Quote'
-  )
+  try {
+    const response = await _fetchGet(
+      chainId,
+      `/markets/${toApiAddress(checkedBaseToken, chainId)}-${toApiAddress(
+        checkedQuoteToken,
+        chainId
+      )}/${kind}/${amount}`
+    )
+
+    return _handleQuoteResponse(response)
+  } catch (error) {
+    throw new QuoteError(UNHANDLED_QUOTE_ERROR)
```

Comment:
Does this work as intended? `_handleQuoteResponse` can throw an specific `QuoteError` that if that happens, it will be captured by this catch and converted into a "generic" one. Or maybe im missing sth

Reply:
good fucking catch, this logic would just rethrow the generic unhandled error. fixed it.

Reply:
:facepalm: